defect-tracker-server/src/main/java/com/defect/tracker/controller/DefectController.java

```diff
@@ -51,31 +92,72 @@
 
 	@PutMapping(value = EndpointURI.DEFECT)
 	public ResponseEntity<Object> editDefectById(@RequestBody DefectDto defectDto) {
-		if (defectService.existsDefectById(defectDto.getId())) {
-			if (defectService.isDefectExistsById(defectDto.getId())) {
-				return new ResponseEntity<>(new ValidationFailureResponse(ValidationConstance.DEFECT_EXISTS,
-						validationFailureStatusCodes.getDefectExistsById()), HttpStatus.BAD_REQUEST);
+		if (!defectService.existsDefectById(defectDto.getId())) {
+			return new ResponseEntity<>(new ValidationFailureResponse(ValidationConstance.DEFECT_NOT_EXISTS_BY_ID,
+					validationFailureStatusCodes.getDefectExistsById()), HttpStatus.BAD_REQUEST);
+		}
+		if (!moduleService.existsModuleById(defectDto.getModuleId())) {
+			return new ResponseEntity<>(new ValidationFailureResponse(ValidationConstance.MODULE_NOT_EXISTS,
+					validationFailureStatusCodes.getModuleById()), HttpStatus.BAD_REQUEST);
+		}
+		if (!projectService.existsProjectById(defectDto.getProjectId())) {
+			return new ResponseEntity<>(new ValidationFailureResponse(ValidationConstance.PROJECT_NOT_EXISTS,
+					validationFailureStatusCodes.getProjectExistsById()), HttpStatus.BAD_REQUEST);
+		}
+		if (!employeeService.isIdAlreadyExists(defectDto.getAssignedById())) {
+			return new ResponseEntity<>(new ValidationFailureResponse(ValidationConstance.ASSIGNED_BY_ID_NOT_EXISTS,
+					validationFailureStatusCodes.getAssignedByExistsById()), HttpStatus.BAD_REQUEST);
+		}
+		if (!employeeService.isIdAlreadyExists(defectDto.getAssignedToId())) {
+			return new ResponseEntity<>(new ValidationFailureResponse(ValidationConstance.ASSIGNED_TO_ID_NOT_EXISTS,
+					validationFailureStatusCodes.getAssignedToExistsById()), HttpStatus.BAD_REQUEST);
+		}
+		if (defectService.getDefectById(defectDto.getId()).equals(defectDto.getStatus())) {
```

Comment:
can you please recheck

Reply:
I checked..
Is validation check have any errors??